kvdb/sqlite/db.go

```diff
@@ -0,0 +1,85 @@
+//go:build kvdb_sqlite
+// +build kvdb_sqlite
+
+package sqlite
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/btcsuite/btcwallet/walletdb"
+	"github.com/lightningnetwork/lnd/kvdb/common_sql"
+	_ "modernc.org/sqlite" // Register relevant drivers.
+)
+
+const (
+	// sqliteOptionPrefix is the string prefix sqlite uses to set various
+	// options. This is used in the following format:
+	//   * sqliteOptionPrefix || option_name = option_value.
+	sqliteOptionPrefix = "_pragma"
+
+	// sqliteTxLockImmediate is dsn option used to ensure that write
+	// transactions are started immediately.
+	sqliteTxLockImmediate = "_txlock=immediate"
```

Comment:
I wasn't able to find this in the official sqlite pragma set: https://www.sqlite.org/pragma.html

AFAICT, this is the same as using `BEGIN IMMEDIATE` for transactions instead of just `BEGIN`. From my reading `IMMEDIATE` is better when you _know_ your transaction will be a write (so it doesn't need to grab the write locks), while `BEGIN` is better for optimistic reads.